mlrun/datastore/__init__.py

```diff
@@ -106,23 +109,57 @@ def get_stream_pusher(stream_path: str, **kwargs):
 
     :param stream_path:        path/url of stream
     """
-
-    kafka_brokers = get_kafka_brokers_from_dict(kwargs)
-    if stream_path.startswith("kafka://") or kafka_brokers:
-        topic, brokers = parse_kafka_url(stream_path, kafka_brokers)
-        return KafkaOutputStream(topic, brokers, kwargs.get("kafka_producer_options"))
-    elif stream_path.startswith("http://") or stream_path.startswith("https://"):
-        return HTTPOutputStream(stream_path=stream_path)
-    elif "://" not in stream_path:
-        return OutputStream(stream_path, **kwargs)
-    elif stream_path.startswith("v3io"):
-        endpoint, stream_path = parse_path(stream_path)
-        endpoint = kwargs.pop("endpoint", None) or endpoint
-        return OutputStream(stream_path, endpoint=endpoint, **kwargs)
-    elif stream_path.startswith("dummy://"):
-        return _DummyStream(**kwargs)
+    if stream_path.startswith("ds://"):
+        from mlrun.datastore.datastore_profile import (
+            DatastoreProfileKafkaSource,
+            DatastoreProfileKafkaTarget,
+            DatastoreProfileV3io,
+        )
+
+        datastore_profile = mlrun.datastore.datastore_profile.datastore_profile_read(
+            stream_path
+        )
```

Comment:
If we get here, the flow is:
1. We have the profile or read it
2. We generate `stream_uri` with it through `get_stream_path`
3. We call `get_stream_pusher` with `stream_uri`
4. We read the profile again

This duplicate profile reading can be spared

Reply:
The stream_uri must be retained to maintain serialization compatibility, as it is embedded within the generated graph structures.